### PR TITLE
authz: add Perforce authz provider

### PR DESCRIPTION
**Pull request by unknwon**
TBD

Part of #16705

### CODE REVIEW

File: enterprise/cmd/precise-code-intel-worker/main.go [hunk 1]
```diff
@@ -125,7 +125,12 @@ func mustInitializeDB() *sql.DB {
 	ctx := context.Background()
 	go func() {
 		for range time.NewTicker(5 * time.Second).C {
-			allowAccessByDefault, authzProviders, _, _ := eiauthz.ProvidersFromConfig(ctx, conf.Get(), database.GlobalExternalServices)
+			allowAccessByDefault, authzProviders, _, _ := eiauthz.ProvidersFromConfig(
```
Comment (efritz) [inline enterprise/cmd/precise-code-intel-worker/main.go:128]: Is there a (future) way to not require this in the worker as it only needs access to `lsif_` tables in the frontend db?

Comment (unknwon) [reply enterprise/cmd/precise-code-intel-worker/main.go:128]: Authz blocking shouldn't get in the way if `database.RepoStore` is not used at all 🤔

Comment (unknwon) [reply enterprise/cmd/precise-code-intel-worker/main.go:128]: What's the easiest way to verify code-intel-worker still functioning if I want to try removing this block of code?

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/main.go:128]: Oh nevermind, the worker has to resolve repo/commits at the beginning of the job.

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/main.go:128]: > What's the easiest way to verify code-intel-worker still functioning

You can run the integration test in internal/cmd/precise-code-intel-tester (upload command is enough). But I remember now that we do hit some repo store methods.

**Pull request closed without merging**